activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,7 @@
+*   Fix ActiveRecord query method when given Date data-type
```

Comment:
It also happens any other types if a type returns a casted value not have `to_i` (e.g. `:json` type).

https://github.com/rails/rails/blob/0ee96d13de29680e148ccb8e5b68025f29fd091c/activerecord/test/models/company.rb#L16

```ruby
Company.new(metadata: {}).metadata?
# => NoMethodError: undefined method `to_i' for {}:Hash
```

Reply:
"query method" is a little ambiguous, can you reword to "query attribute method"?

Reply:
> It also happens on any other types if a type returns a casted value not have `to_i` (e.g. `:json` type)

Should I add test cases that user define a type does not respond to to_i method?

Reply:
That sounds good to me, since "when given Date data-type" is a part of what we fixes.